library/src/main/java/io/realm/RealmBasedRecyclerViewAdapter.java

```diff
@@ -423,8 +443,26 @@ private void updateRowWrappers() {
             final long headerIndex = realmResults.getTable().getColumnIndex(headerColumnName);
             int i = 0;
             for (RealmModel result : realmResults) {
-                String rawHeader = ((RealmObjectProxy) result)
-                        .realmGet$proxyState().getRow$realm().getString(headerIndex);
+                Object rawHeader = null;
+                RealmFieldType fieldType = ((RealmObjectProxy) result)
+                        .realmGet$proxyState().getRow$realm().getColumnType(headerIndex);
+
+                if (fieldType == RealmFieldType.STRING) {
```

Comment:
Same here. Use if / else if statements and throw an exception for the default else case.